packages/ra-ui-materialui/src/field/ArrayField.tsx

```diff
@@ -88,78 +117,43 @@ const initialState = {
  *     )
  *     TagsField.defaultProps = { addLabel: true };
  */
-export class ArrayField extends Component<
-    FieldProps & InjectedFieldProps,
-    State
-> {
-    constructor(props: FieldProps & InjectedFieldProps) {
-        super(props);
-        this.state = props.record
-            ? this.getDataAndIds(props.record, props.source, props.fieldKey)
-            : initialState;
-    }
-
-    componentWillReceiveProps(
-        nextProps: FieldProps & InjectedFieldProps,
-        prevProps: FieldProps & InjectedFieldProps
-    ) {
-        if (nextProps.record !== prevProps.record) {
-            this.setState(
-                this.getDataAndIds(
-                    nextProps.record,
-                    nextProps.source,
-                    nextProps.fieldKey
-                )
-            );
-        }
-    }
-
-    getDataAndIds(record: object, source: string, fieldKey: string) {
+export const ArrayField: FunctionComponent<
+    FieldProps & InjectedFieldProps & State
+> = ({
+    addLabel,
+    basePath,
+    children,
+    record,
+    sortable,
+    source,
+    fieldKey,
+    ...rest
+}) => {
+    const [ids, setIds] = useState(() => {
         const list = get(record, source);
-        if (!list) {
-            return initialState;
-        }
+        if (!list) return [];
         return fieldKey
-            ? {
-                  data: list.reduce((prev, item) => {
-                      prev[item[fieldKey]] = item;
-                      return prev;
-                  }, {}),
-                  ids: list.map(item => item[fieldKey]),
-              }
-            : {
-                  data: list.reduce((prev, item) => {
-                      prev[JSON.stringify(item)] = item;
-                      return prev;
-                  }, {}),
-                  ids: list.map(JSON.stringify),
-              };
-    }
+            ? list.map(item => item[fieldKey])
+            : list.map(JSON.stringify);
+    });
+    const [data, setData] = useState(ids);
```

Comment:
This seems weird to me. According to the previous code, this should be something similar to the other useState code

Reply:
Ok.